lib/internal/crypto/random.js

```diff
@@ -281,10 +282,86 @@ function getRandomValues(data) {
   return data;
 }
 
+// Implements an RFC 4122 version 4 random UUID.
+// To improve performance, random data is generated in batches
+// large enough to cover kBatchSize UUID's at a time. The uuidData
+// and uuid buffers are reused for each. Each call to randomUUID()
+// consumes 16 bytes from the buffer.
+
+const kHexDigits = [
+  48, 49, 50, 51, 52, 53, 54, 55,
+  56, 57, 97, 98, 99, 100, 101, 102
+];
+
+const kBatchSize = 128;
+let uuidData;
+let uuid;
+let uuidBatch = 0;
+
+function randomUUID() {
+  if (uuidData === undefined) {
+    uuidData = new Uint8Array(16 * kBatchSize);
+    uuid = Buffer.alloc(36, '-');
+    uuid[14] = 52; // '4'
+  }
+
+  if (uuidBatch === 0) randomFillSync(uuidData);
+  const base = 16 * uuidBatch;
+  uuidBatch = (uuidBatch + 1) % kBatchSize;
+
+  // Variant byte: 10xxxxxx (variant 1)
+  uuidData[base + 8] = (uuidData[base + 8] & 0x3f) | 0x80;
+
+  // This function is structured the way it is for performance.
+  // The uuid buffer stores the serialization of the random
+  // bytes from uuidData.
+  // xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+  let n = base;
+  uuid[0] = kHexDigits[uuidData[n] >> 4];
+  uuid[1] = kHexDigits[uuidData[n++] & 0xf];
+  uuid[2] = kHexDigits[uuidData[n] >> 4];
+  uuid[3] = kHexDigits[uuidData[n++] & 0xf];
+  uuid[4] = kHexDigits[uuidData[n] >> 4];
+  uuid[5] = kHexDigits[uuidData[n++] & 0xf];
+  uuid[6] = kHexDigits[uuidData[n] >> 4];
+  uuid[7] = kHexDigits[uuidData[n++] & 0xf];
+  // -
+  uuid[9] = kHexDigits[uuidData[n] >> 4];
+  uuid[10] = kHexDigits[uuidData[n++] & 0xf];
+  uuid[11] = kHexDigits[uuidData[n] >> 4];
+  uuid[12] = kHexDigits[uuidData[n++] & 0xf];
+  // -
+  // uuid[14] is set already...
+  uuid[15] = kHexDigits[uuidData[n++] & 0xf];
+  uuid[16] = kHexDigits[uuidData[n] >> 4];
+  uuid[17] = kHexDigits[uuidData[n++] & 0xf];
+  // -
+  uuid[19] = kHexDigits[uuidData[n] >> 4];
```

Comment:
```suggestion
  uuid[19] = kHexDigits[(uuidData[n] >> 4) & 3 | 8]; // variant: 10xx
```

Reply:
I'd rather keep this separate as I think it makes it slightly more difficult to read and doesn't impact perf